librz/reg/profile.c

```diff
@@ -1,245 +1,452 @@
 // SPDX-FileCopyrightText: 2009-2020 pancake <pancake@nopcode.org>
 // SPDX-License-Identifier: LGPL-3.0-only
 
+#include <rz_list.h>
+#include <rz_util/rz_log.h>
+#include <rz_util/rz_str.h>
 #include <rz_reg.h>
 #include <rz_util.h>
+#include <rz_util/rz_assert.h>
 #include <rz_lib.h>
+#include <string.h>
 
-static const char *parse_alias(RzReg *reg, char **tok, const int n) {
-	if (n == 2) {
-		int role = rz_reg_get_name_idx(tok[0] + 1);
-		return rz_reg_set_name(reg, role, tok[1])
-			? NULL
-			: "Invalid alias";
+static void rz_reg_profile_def_free(RzRegProfileDef *def) {
+	if (!def) {
+		return;
 	}
-	return "Invalid syntax";
+	if (def->name) {
```

Comment:
Just `free(def->name)` should be enough, all implementations handle `free(NULL)` properly.